packages/next/src/client/app-dir/link.tsx

```diff
@@ -247,6 +250,7 @@ function onNavigationIntent(element: HTMLAnchorElement | SVGAElement) {
   }
   // Prefetch the link on hover/touchstart.
   if (instance !== undefined) {
+    instance.wasHoveredOrTouched = true
```

Comment:
does/should this encompass keyboard focus as well?

Reply:
I think so, maybe. I'll add to the project tracker.